arb_os/chainParameters.mini

```diff
@@ -37,7 +43,40 @@ public impure func chainParams_gotParamsMessage(sender: address, data: Marshalle
             baseStake: bytearray_get256(ba, 3*32),
             stakingToken: address(bytearray_get256(ba, 4*32)),
             owner: address(bytearray_get256(ba, 5*32)),
+            isProChain: false,
         });
+
+        // now read and process initialization options
+        if let Some(stream) = bytestream_skipBytes(bytestream_new(ba), 5*32) {
+            loop {
+                if let Some(ustr2) = doInitializationOption(stream) {
+                    stream = ustr2;
+                } else {
+                    return;
+                }
+            }
+        }
+    }
+}
+
+impure func doInitializationOption(stream: ByteStream) -> option<ByteStream> {
+    // process one initialization option
+    // on success, return Some(updated stream)
+    // on end-of-stream, return None
+
+    let (ustr, optionId) = bytestream_get64(stream)?;
+    stream = ustr;
+    let (ustr, numPayloadBytes) = bytestream_get64(stream)?;
+    stream = ustr;
+    if (optionId == 0) {   // option: premium chain
+        if (numPayloadBytes != 0) {
+            return None;
```

Comment:
Currently this causes the parsing of options to terminate. It might be better to instead just ignore this option and go to the next one

Reply:
Changed this to ignore the option and move to the next one if size is unexpected.